pkg/data/redis.go

```diff
@@ -0,0 +1,458 @@
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+
+	api "kubedb.dev/apimachinery/apis/kubedb/v1alpha2"
+	cs "kubedb.dev/apimachinery/client/clientset/versioned"
+	"kubedb.dev/cli/pkg/data/redisutil"
+	_ "kubedb.dev/db-client-go/redis"
+
+	"github.com/spf13/cobra"
+	shell "gomodules.xyz/go-sh"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"k8s.io/klog/v2"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+type redisOpts struct {
+	db       *api.Redis
+	dbClient *cs.Clientset
+
+	errWriter *bytes.Buffer
+}
+type MasterNode struct {
+	host string
+	slot int64
+}
+
+var dataInsertScript = `
+for i = 1, ARGV[1], 1 do
+	local str = tostring({}):sub(10)
+    redis.call("SET", "kubedb:"..str.."{"..ARGV[2].."}-key"..i, str)
+end
+
+return "Success!"
+`
+
+var dataDeleteScript = `
+local cursor = 0
+local calls = 0
+local dels = 0
+repeat
+    local result = redis.call('SCAN', cursor, 'MATCH', ARGV[1])
+    calls = calls + 1
+    for _,key in ipairs(result[2]) do
+        redis.call('DEL', key)
+        dels = dels + 1
+    end
+    cursor = tonumber(result[1])
+until cursor == 0
+
+return "Success!"
+`
+
+const (
+	redisKeyPrefix         = "kubedb"
+	successfulScriptReturn = "Success!"
+)
+
+func InsertRedisDataCMD(f cmdutil.Factory) *cobra.Command {
+	var (
+		dbName string
+		rows   int
+	)
+
+	rdInsertCmd := &cobra.Command{
+		Use: "redis",
+		Aliases: []string{
+			"rd",
+		},
+		Short:   "Insert data to a redis database",
+		Long:    `Use this cmd to insert data into a redis database`,
+		Example: `kubectl dba insert redis -n demo sample-redis --rows 500`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) == 0 {
+				log.Fatal("Enter redis object's name as an argument")
+			}
+			dbName = args[0]
+
+			namespace, _, err := f.ToRawKubeConfigLoader().Namespace()
+			if err != nil {
+				klog.Error(err, "failed to get current namespace")
+			}
+
+			opts, err := newRedisOpts(f, dbName, namespace)
+			if err != nil {
+				log.Fatalln(err)
+			}
+
+			err = opts.insertDataInDatabase(rows)
+			if err != nil {
+				log.Fatal(err)
+			}
+		},
+	}
+	rdInsertCmd.Flags().IntVarP(&rows, "rows", "r", 10, "rows in ")
+
+	return rdInsertCmd
+}
+
+func (opts *redisOpts) insertDataInDatabase(rows int) error {
+	if opts.db.Spec.Mode == api.RedisModeCluster {
+		return opts.insertDataInRedisCluster(rows)
+	}
+
+	redisCommand := []interface{}{
+		"eval", dataInsertScript, "0", fmt.Sprintf("%d", rows), "hash",
+	}
+	output, err := opts.execCommand("", redisCommand)
+	if err != nil {
+		return err
+	}
+	if output != successfulScriptReturn {
+		fmt.Printf("Error. Can not insert data in master node. Output: %s\n", output)
+	}
+	fmt.Printf("\n%d keys inserted in redis database %s/%s successfully\n", rows, opts.db.Namespace, opts.db.Name)
+	return nil
+}
+
+func (opts *redisOpts) insertDataInRedisCluster(rows int) error {
+	var slotKey map[int64]string
+	err := json.Unmarshal(redisutil.ClusterSlotKeys, &slotKey)
```

Comment:
Why do we need a seperate file for insertion ?  Can't we generate random key-value pairs in runtime ?

Reply:
It will not be efficient to generate it in runtime